libraries/AP_UAVCAN/AP_UAVCAN.h

```diff
@@ -336,21 +235,68 @@ class AP_UAVCAN : public AP_CANDriver, public AP_ESC_Telem_Backend {
 
     static HAL_Semaphore _telem_sem;
 
+    // node status send
+    uint32_t _node_status_last_send_ms;
+
     // safety status send state
     uint32_t _last_safety_state_ms;
 
     // notify vehicle state
     uint32_t _last_notify_state_ms;
+    uavcan_protocol_NodeStatus node_status_msg;
+
+    CanardInterface canard_iface;
+    Canard::Publisher<uavcan_protocol_NodeStatus_cxx_iface> node_status{canard_iface};
```

Comment:
what is the memory cost of these publishers when DroneCAN is not enabled?

Reply:
There's no cost, we allocate the class only when we use it. Previously, the publishers were static pointers, so technically the memory cost is less than before.